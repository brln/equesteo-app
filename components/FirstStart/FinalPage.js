import React, { PureComponent } from 'react'
import {
  Dimensions,
  StyleSheet,
  Text,
  View
} from 'react-native'

import { brand } from '../../colors'
import { logRender } from '../../helpers'
import Button from '../Button'

const { height, width } = Dimensions.get('window')

export default class FinalPage extends PureComponent {
  constructor(props) {
    super(props)
    this.done = this.done.bind(this)
  }

  done () {
    this.props.done(true)
  }

  render() {
    logRender('FinalPage')

    return (
      <View style={{flex: 1}}>
        <View style={{flex: 1}}>
          <View style={{flex: 1}}>
            <Text style={{
              fontSize: 30,
              fontFamily: 'Montserrat-Regular',
              color: 'black',
              textAlign: 'center'
            }}>
              You're all set!
            </Text>
          </View>
          <View style={{flex: 4}}>
            <View>
              <Text style={{textAlign: 'center', fontSize: 20, padding: 20}}>If you need any help, have ideas or concerns about the app, or just want to talk horses, email me!</Text>
            </View>
            <View>
              <Text style={{textAlign: 'center', fontSize: 20}}>nicole@equesteo.com</Text>
              <View style={{paddingTop: 20, marginRight: 40, marginLeft: 40}}>
                <Button
                  color={brand}
                  text={"Got It"}
                  onPress={this.done}
                />
              </View>
            </View>
          </View>
        </View>

      </View>
    )
  }
}

const styles = StyleSheet.create({
});
