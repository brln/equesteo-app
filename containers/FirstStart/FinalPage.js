import React, { PureComponent } from 'react'
import { connect } from 'react-redux'

import {
  Text,
  View
} from 'react-native'

import { brand } from '../../colors'
import { logRender } from '../../helpers'
import Button from '../../components/Button'
import EqNavigation from '../../services/EqNavigation'
import {
  deleteUnpersistedHorse,
  userUpdated,
} from '../../actions/standard'
import functional from '../../actions/functional'
import Wrapper from '../../components/FirstStart/Wrapper'


class FinalPage extends PureComponent {
  static options() {
    return {
      topBar: {
        background: {
          color: 'transparent',
        },
        elevation: 0,
        backButton: {
          color: brand,
        },
      },
      layout: {
        orientation: ['portrait']
      }
    }
  }

  constructor(props) {
    super(props)
    this.done = this.done.bind(this)
  }

  done () {
    this.props.dispatch(userUpdated(this.props.user.set('finishedFirstStart', true)))
    this.props.dispatch(functional.persistUserUpdate(this.props.user.get('_id'), [], false))

    EqNavigation.popToRoot(this.props.componentId).then(() => {
      if (this.props.skippedHorse) {
        this.props.dispatch(deleteUnpersistedHorse(this.props.horseID, this.props.horseUserID))
      }
      this.props.dispatch(functional.doSync())
    }).catch(() => {})
  }

  render() {
    logRender('FinalPage')
    return (
      <Wrapper>
        <View style={{flex: 1}}>
          <View style={{flex: 1}}>
            <View style={{flex: 1}}>
              <Text style={{
                fontSize: 30,
                fontFamily: 'Montserrat-Regular',
                color: 'black',
                textAlign: 'center'
              }}>
                You're all set!
              </Text>
            </View>
            <View style={{flex: 4}}>
              <View>
                <Text style={{textAlign: 'center', fontSize: 20, padding: 20}}>If you need any help, have ideas or concerns about the app, or just want to talk horses, email me!</Text>
              </View>
              <View>
                <Text style={{textAlign: 'center', fontSize: 20}}>nicole@equesteo.com</Text>
                <View style={{paddingTop: 20, marginRight: 40, marginLeft: 40}}>
                  <Button
                    color={brand}
                    text={"Got It"}
                    onPress={this.done}
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </Wrapper>
    )
  }
}

function mapStateToProps (state, passedProps) {
  const localState = state.get('localState')
  const pouchState = state.get('pouchRecords')
  const horseID = localState.getIn(['firstStartHorseID', 'horseID'])
  const horseUserID = localState.getIn(['firstStartHorseID', 'horseUserID'])
  return {
    horseID,
    horseUserID,
    skippedHorse: passedProps.skippedHorse,
    user: pouchState.getIn(['users', localState.get('userID')]),
  }
}

export default connect(mapStateToProps)(FinalPage)
