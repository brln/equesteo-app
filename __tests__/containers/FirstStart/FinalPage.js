import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import FinalPage from '../../../containers/FirstStart/FinalPage'


const mockStore = configureStore([thunk])

test('render', () => {
  const store = mockStore(fromJS({
    localState: {},
    pouchRecords: {},
  }))

  const tree = renderer.create(
    <Provider store={store}>
      <FinalPage />
    </Provider>)
  expect(tree.toJSON()).toMatchSnapshot()
})

