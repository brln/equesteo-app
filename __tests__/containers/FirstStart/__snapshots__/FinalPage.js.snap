// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View>
  <View
    style={
      Object {
        "height": 1278,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 3,
          "justifyContent": "flex-end",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/firstStart.jpg",
          }
        }
        style={
          Object {
            "height": 750,
            "resizeMode": "cover",
            "width": 750,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "column",
            "paddingTop": 20,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "width": 60,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "black",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 30,
                  "textAlign": "center",
                }
              }
            >
              You're all set!
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 4,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "padding": 20,
                    "textAlign": "center",
                  }
                }
              >
                If you need any help, have ideas or concerns about the app, or just want to talk horses, email me!
              </Text>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                nicole@equesteo.com
              </Text>
              <View
                style={
                  Object {
                    "marginLeft": 40,
                    "marginRight": 40,
                    "paddingTop": 20,
                  }
                }
              >
                <View
                  accessible={true}
                  clickable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#5bc0eb",
                      "borderColor": undefined,
                      "borderRadius": 10,
                      "borderWidth": null,
                      "marginTop": 10,
                      "opacity": 1,
                      "paddingBottom": 10,
                      "paddingTop": 10,
                      "width": undefined,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "paddingLeft": 20,
                          "paddingRight": 20,
                          "textAlign": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    Got It
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
